### PR TITLE
loading buttons

### PR DESCRIPTION
**Pull request by emma-floored**
From Ben originally! This PR is not ready yet, but I wanted to open it anyway to help me keep track of the loading button specific differences.

### CODE REVIEW

**Comment by emma-floored** [comment]
@MagdalenaHenke @Jinsoo94 ok - updated this. A few things...

- I made some icon related changes as I was working on the indicator icons for loading buttons
  - mainly, Icon -> SvgIcon
- I got rid of the loading button component 😱 I think it is maybe something that makes more sense to leave to the apps to implement? but I did an example with example code.

File: packages/blocks-react/components/buttons/Button.jsx [hunk 1]
```diff
@@ -19,20 +19,21 @@ const TYPE_CLASS_MAP = {
...
   );
 });
 
 Button.propTypes = {
   className: PropTypes.string,
   style: PropTypes.object,
-  color: PropTypes.string,
-  type: PropTypes.string
+  color: PropTypes.oneOf([ 'primary', 'dark', 'secondary', 'danger' ]),
+  type: PropTypes.oneOf([ 'default', 'icon' ])
```
Comment (Jinsoo94) [inline packages/blocks-react/components/buttons/Button.jsx]: nice!

**Comment by Jinsoo94** [comment]
tested locally. Looks good! 

hmm why up to individual apps? because some apps might not want to use Blocks loading icon?

**Comment by emma-floored** [comment]
@Jinsoo94 on Ben's old PR there was a concern about the state of buttons and possibly wanting to use "loading" type buttons in other situations (ex. without the 'loading' state at all, or maybe they should show errors, etc) - https://github.com/floored/blocks/pull/111#discussion_r259083690

I feel like this solution gives the most flexibility.

**Comment by Jinsoo94** [comment]
aha makes sense. other than that looks good to me!

**Comment by MagdalenaHenke** [comment]
Ooooh was goona look at this now - but sorry I caused you conflicts

**Comment by emma-floored** [comment]
@MagdalenaHenke conflicts should be resolved!

**Comment by bleveque** [comment]
going to close this one for now, since it's not a current high priority, but we could always decide to just reopen when someone picks up this ticket: https://app.clubhouse.io/blocks/story/47126/loading-button-example

**Pull request closed without merging**